docs/proposals/zeroconf.md

```diff
@@ -0,0 +1,142 @@
+
+## Broker interfacing
+
+Upon detection of ZeroConf hosts and services, the Akri ZeroConf protocol creates "twins" for each service using the provided, sample broker. A more complete rendition of the `freddie` service could be:
+
+```YAML
+{
+    name: "freddie",
+    kind: "_example._tcp",
+    domain: "local",
+    host_name: "freddie.local",
+    address: "192.168.1.100", port: 8888,
+    txt: ...
+}
+```
+
+Each Broker Instance is configured with environment variables corresponding to the above value:
+
+```bash
+AKRI_ZEROCONF_DEVICE_KIND=_example._tcp
+AKRI_ZEROCONF=zeroconf
+AKRI_ZEROCONF_DEVICE_HOST=freddie.local
+AKRI_ZEROCONF_DEVICE_NAME=freddie
+AKRI_ZEROCONF_DEVICE_PORT=8888
+AKRI_ZEROCONF_DEVICE_ADDR=192.168.1.100
+```
+
+Currently the service's TXT records are not provided.
+
+The Broker sample enumerates these values to standard output every 5 seconds.
+
+In practice, a Kubernetes application would use this data to identify services and invoke them.
+
+## Security Considerations
+
+The Akri Agent is only able to browse services accessible to its cluster's hosts.
+
+There are no security considerations for service (mDNS) browsing. This functionality is akin to DNS lookups.
+
+Accessing services that are found by service browsing *may* require the provision of credentials. This would be an implementation detail of an Akri Broker and is not considered further here.
+
+## Outstanding Questions
+
++ What would a generic Akri ZeroConf Broker do? In practice, the application developer would likely wish to implement the Broker for their specific application.
+
+The limit of a generic Akri ZeroConf Broker is to enumerate services that it discovers and this is demonstrated by a sample Broker included in the ZeroConf Protocol implementation. In practice, a ZeroConf Broker would need to be aware of the implementation(s) of the service that it "twins". There is a limitation on the transport protocols that are accessible to Akri (discussed below) but, for the permitted transport protocols, there are potentially limitless service types and implementation details and these are all potentially accessible to an Akri Broker using this ZeroConf protocol implementation.
+
++ How to treat support for Kubernetes-supporting service types (TCP, UDP, SCTP)?
+
+Kubernetes supports [TCP, UDP, SCTP](https://kubernetes.io/docs/concepts/services-networking/service/#protocol-support) transport protocols. Service discovery supports other transport protocols. Because Akri is dependent on Kubernetes, the Akri ZeroConf Protocol implementation only supports these 3 transport protocols too.
+
+## Feature Requests
+
++ Discovery should differentiate between services that are supportable (!) by Kubernetes (TCP, UDP, SCTP) and those that aren't
+
+The [`zeroconf-filter`](https://github.com/DazWilkin/akri-pest) crate used by the ZeroConf Protocol implementation only permits: TCP, UDP and SCTP.
```

Comment:
Same comment as above. I dont see why this needs to be limited

Reply:
Fixed: 5b0fed2